app/controllers/orders_controller.rb

```diff
@@ -33,8 +34,36 @@ def index
                           .per(Settings.length.digit_8)
   end
 
+  def cancel
+    if @order.pending?
+      @order.cancel!
+      flash[:success] = t "order.cancel_success"
+    else
+      flash[:danger] = t "order.cancel_failed"
+    end
+  rescue ActiveRecord::RecordInvalid
+    flash[:danger] = t "order.cancel_failed"
+  ensure
+    redirect_back(fallback_location: root_path)
+  end
+
   private
 
+  def check_owner_order
+    return if @order.user_id == current_user.id
+
+    flash[:danger] = t "order.access_denied"
+    redirect_to static_pages_home_path
+  end
+
+  def load_order
+    @order = Order.find_by id: params[:order_id]
```

Comment:
hí hí
chỗ này á, ko cần check owner order đâu chỉ cần change code này thành như bên dưới, method `check_owner_order` cũng ko cần luôn 


```suggestion
    @order = current_user.orders.find_by id: params[:order_id]
```